Sources/_CryptoExtras/ChaCha20CTR/BoringSSL/ChaCha20CTR_boring.swift

```diff
@@ -0,0 +1,56 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftCrypto open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftCrypto project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.md for the list of SwiftCrypto project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+@_implementationOnly import CCryptoBoringSSL
+@_implementationOnly import CCryptoBoringSSLShims
+import Crypto
+@_implementationOnly import CryptoBoringWrapper
+import Foundation
+
+enum OpenSSLChaCha20CTRImpl {
+    static func encrypt<M: DataProtocol, N: ContiguousBytes>(key: SymmetricKey, message: M, counter: UInt32, nonce: N) throws -> Data {
+        guard key.bitCount == Insecure.ChaCha20CTR.keyBitsCount else {
+            throw CryptoKitError.incorrectKeySize
+        }
+
+        var ciphertext = Array<UInt8>(repeating: 0, count: message.count)
+
+        key.withUnsafeBytes { keyPtr in
+            nonce.withUnsafeBytes { noncePtr in
+                message.withContiguousStorageIfAvailable { plaintext in
```

Comment:
This call isn't quite right. We need to do a bit of a dance here. I recommend looking at the way the `seal` function works, where there is an outer layer that optionally flattens the `DataProtocol` before calling an inner layer that is generic over `ContiguousBytes`. That's what we want to do here.

Reply:
Okay hopefully https://github.com/apple/swift-crypto/pull/169/commits/17d166b425bbdf5b0a4d9fba30a55ee19bb58635 implements this correctly?